pkg/apis/app/v1alpha1/kogitobuild_types.go

```diff
@@ -0,0 +1,236 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// KogitoBuildType describes the build types supported by the KogitoBuild CR
+type KogitoBuildType string
+
+const (
+	// BinaryBuildType builds takes an uploaded binary file already compiled and creates a Kogito Runtime Service image from it
+	BinaryBuildType KogitoBuildType = "Binary"
+	// RemoteSourceBuildType builds pulls the source code from a Git repository, builds the binary and then the final Kogito Runtime Service image
+	RemoteSourceBuildType KogitoBuildType = "RemoteSource"
+	// LocalSourceBuildType builds takes an uploaded resource files such as DRL (rules), DMN (decision) or BPMN (process), builds the binary and the final Kogito Runtime Service image
+	LocalSourceBuildType KogitoBuildType = "LocalSource"
+)
+
+// KogitoBuildSpec defines the desired state of KogitoBuild
+type KogitoBuildSpec struct {
+
+	// Sets the type of build that this instance will handle:
+	// Binary - takes an uploaded binary file already compiled and creates a Kogito Runtime Service image from it
+	// RemoteSource - pulls the source code from a Git repository, builds the binary and then the final Kogito Runtime Service image
+	// LocalSource - takes an uploaded resource file such as DRL (rules), DMN (decision) or BPMN (process), builds the binary and the final Kogito Runtime Service image
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="DisableIncremental Builds"
+	// +kubebuilder:validation:Enum=Binary;RemoteSource;LocalSource
+	Type KogitoBuildType `json:"type"`
+
+	// DisableIncremental indicates that source to image builds should NOT be incremental. Defaults to false.
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="DisableIncremental Builds"
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.x-descriptors="urn:alm:descriptor:com.tectonic.ui:booleanSwitch"
+	// +optional
+	DisableIncremental bool `json:"disableIncremental,omitempty"`
+
+	// Environment variables used during build time
+	// +listType=atomic
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="Build Env Variables"
+	// +optional
+	Envs []corev1.EnvVar `json:"envs,omitempty"`
+
+	// Information about the git repository where the Kogito Service source code resides.
+	// Ignored for binary builds
+	// +optional
+	GitSource GitSource `json:"gitSource,omitempty"`
```

Comment:
Does GitSource support private GIT repositories? In this case, we would need to provide the SSH credentials. If not, we should raise a ticket for this.

Reply:
Yes, any kind of git repos. Well put, we can definitely add this support if BuildConfig supports it. Could you please open a JIRA for me?

Reply:
Using git@github.com:kiegroup/kogito-examples.git is not working, it fails with:

{code}
Error: stat git@github.com:Sgitario/kogito-examples.git: no such file or directory
{code}

But without SSH, it works fine setting the username and password like this: https://username:password@github.com/username/repo_name.git.

Should I raise an issue to support SSH?

Reply:
Please, raise and I'll take a look. Even if it's just an update in the docs to use the format with `username:password@`.